### PR TITLE
Add wrap for SQLite3.

### PR DESCRIPTION
**Pull request by jkuebart**

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    sqlite_args += ['-DSQLITE_API=__declspec(dllexport)']
+endif
+
+includes = include_directories('.')
+libsqlite3 = library(
+    'sqlite3',
+    'sqlite3.c',
+    c_args: sqlite_args,
+    dependencies: [libdl, libm, dependency('threads')],
+    include_directories: includes,
```
Comment (jpakkane) [inline meson.build]: You don't need this for the compilation, current source and build dirs are always in header search path by default.

**Comment by xclaesse** [comment]
@jkuebart @jpakkane We already had a "sqlite.wrap", is there a reason we added sqlite3.wrap instead of updating the other one? From a quick look this port seems more advanced.

**Comment by jkuebart** [comment]
> @jkuebart @jpakkane We already had a "sqlite.wrap", is there a reason we added sqlite3.wrap instead of updating the other one? From a quick look this port seems more advanced.

The existing wrap had a number of problems. First of all, the package is called sqlite3 officially and by most other package managers. Second, the version number of the existing package was incorrect. Both of these prevented using the wrap as a fallback to a locally installed package.

See https://github.com/mesonbuild/wrapdb/issues/80#issuecomment-756992987 for more background.

**Comment by xclaesse** [comment]
Makes sense thanks

**Pull request merged**